### PR TITLE
feat(animation): add animation timing factor global var and match animation timing with calcite base value

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #3491 

## Summary

This will make sure global animation timing class `--calcite-animation-timing` value matches with the `calcite-base` [$transition](https://github.com/Esri/calcite-base/blob/master/dist/_variables.scss#L2) variable value and brings uniformity among components in calcite repo.

### CODE REVIEW

**Comment by benelan** [comment]
> /merge/master

It's `/merge master`

File: src/tests/globalStyles.e2e.ts [hunk 1]
```diff
@@ -44,7 +44,26 @@ describe("global styles", () => {
...
           return window.getComputedStyle(document.body).getPropertyValue("--calcite-animation-timing");
         })
-      ).toBe(" 300ms");
+      ).toBe(" 150ms");
+    });
+
+    it("should set animation duration to 0ms", async () => {
+      const page = await newE2EPage({
+        html: snippet
+      });
```
Comment (jcfranco) [inline src/tests/globalStyles.e2e.ts:53]: For test stability, can you add a `await page.waitForChanges()` call here?

File: src/tests/globalStyles.e2e.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ describe("global styles", () => {
             opacity: opacity
           };
```
Comment (jcfranco) [inline src/tests/globalStyles.e2e.ts:33]: @caripizza Do these tests need to use `notice`? I originally thought they were testing component-specific animation, but it looks like the tests focus on asserting values on the global classes.

File: src/components/calcite-action-pad/calcite-action-pad.scss [hunk 3]
```diff
@@ -1,5 +1,6 @@
 :host {
   @extend %component-host;
+  --calcite-animation-timing: 300ms;
```
Comment (jcfranco) [inline src/components/calcite-action-pad/calcite-action-pad.scss:3]: Aside from focus timing that will go away, I thought all components would rely on the same global timing var. Am I missing something? cc @caripizza @macandcheese

Comment (caripizza) [reply src/components/calcite-action-pad/calcite-action-pad.scss:3]: @jcfranco no I think you have it right. It's just that a few components which reference these global animation utility classes (ex `animate-in-up`) seem like they need the longer 300ms speed for their animations (not the new 150ms speed from updated the global var value). That's why I recommended that Anvesh just reset their values directly above the animation class as he's doing here.

Comment (jcfranco) [reply src/components/calcite-action-pad/calcite-action-pad.scss:3]: >  It's just that a few components which reference these global animation utility classes (ex animate-in-up) seem like they need the longer 300ms speed for their animations

I see. What are the downsides to having all of them use the global var? Would some animations look off? 

We need a way to allow users to override the global variable to tweak or even disable animations. Having component-specific overrides does defeats the purpose. I think @anveshmekala brought this up before. Looping in @asangma + @macandcheese for suggestions.

Comment (caripizza) [reply src/components/calcite-action-pad/calcite-action-pad.scss:3]: These utility classes are using the global var already - is that your concern?
https://github.com/Esri/calcite-components/blob/master/tailwind.config.js#L130

File: src/tests/globalStyles.e2e.ts (hunk 2 above)
Comment (caripizza) [reply src/tests/globalStyles.e2e.ts:33]: Nope, please remove or improve them (these were just tests I added with the [original demo page](https://github.com/Esri/calcite-components/blob/master/src/demos/animation/animation.html)).

It seems more important to test that animations can be disabled globally by setting the timing var to `0s`.

File: src/components/calcite-action-pad/calcite-action-pad.scss (hunk 3 above)
Comment (jcfranco) [reply src/components/calcite-action-pad/calcite-action-pad.scss:3]: My concern having a mixed usage of the global var and some components overriding the duration var. I'd like to have users override a single global var for all animation/transition duration. Let me know if this makes sense. We can have a quick chat later today if I'm misunderstanding the approach.

File: src/tests/globalStyles.e2e.ts [hunk 4]
```diff
@@ -32,19 +31,78 @@ describe("global styles", () => {
...
     });
 
-    it("should have initial --calcite-animation-timing CSS Custom Property value", async () => {
-      const page = await newE2EPage({ html: snippet });
-      expect(
-        await page.evaluate(() => {
-          return window.getComputedStyle(document.body).getPropertyValue("--calcite-animation-timing");
-        })
-      ).toBe(" 300ms");
+    it("should set animation duration to 0ms", async () => {
```
Comment (jcfranco) [inline src/tests/globalStyles.e2e.ts:40]: I'm not sure what this test is covering. `--calcite-animation-timing` is internal, so no need to test it, right? Maybe I'm missing something.

File: src/tests/globalStyles.e2e.ts [hunk 5]
```diff
@@ -32,19 +31,78 @@ describe("global styles", () => {
...
+      const page = await newE2EPage({
+        html: snippet
+      });
+      await page.waitForChanges();
+      await page.$eval("calcite-notice", (element: any) => {
```
Comment (jcfranco) [inline src/tests/globalStyles.e2e.ts:45]: Type the element and avoid using `any` types whenever possible. This applies to similar functions.

File: src/assets/styles/_animation.scss [hunk 6]
```diff
@@ -41,7 +41,11 @@
 }
 
 :root {
-  --calcite-animation-timing: 300ms;
+  --calcite-animation-timing-factor: var(--animation-timing-factor, 1);
```
[the remark this answers is not in the document]
Comment (anveshmekala) [reply src/assets/styles/_animation.scss:44]: Make sense. Will swap them.

File: src/assets/styles/_animation.scss [hunk 7]
```diff
@@ -41,7 +41,11 @@
...
+  --calcite-animation-timing: calc(150ms * var(--calcite-animation-timing-factor));
```
[the remark this answers is not in the document]
Comment (anveshmekala) [reply src/assets/styles/_animation.scss:45]: These are going to be internal for now. We let user's control the animation duration with `--calcite-animation-timing-factor` .

File: src/tests/globalStyles.e2e.ts (hunk 4 above)
Comment (anveshmekala) [reply src/tests/globalStyles.e2e.ts:40]: This test was added prior to the audit. Might have overlooked and  this test makes more sense now considering `--calcite-animation-factor` will be the public.

Comment (anveshmekala) [reply src/tests/globalStyles.e2e.ts:40]: never mind. This test should be removed.

File: src/assets/styles/_animation.scss [hunk 8]
```diff
@@ -41,7 +41,11 @@
 }
 
 :root {
-  --calcite-animation-timing: 300ms;
+  --animation-timing-factor: var(--calcite-animation-timing-factor, 1);
```
Comment (jcfranco) [inline src/assets/styles/_animation.scss:44]: Per our conversation w/ @macandcheese, all root internal CSS props should be namespaced w/ `--calcite-internal-<prop>`. Please rename the following:

* `--animation-timing-factor` -> `--calcite-internal-animation-timing-factor`
* `--calcite-animation-timing-quick` -> `--calcite-internal-animation-timing-quick`
* `--calcite-animation-timing-medium` -> `--calcite-internal-animation-timing-medium`
* `--calcite-animation-timing-slow` -> `--calcite-internal-animation-timing-slow`

Note: `--calcite-animation-timing` stays the same as it is public.

File: src/tests/globalStyles.e2e.ts [hunk 9]
```diff
@@ -32,19 +31,58 @@ describe("global styles", () => {
...
     });
 
-    it("should have initial --calcite-animation-timing CSS Custom Property value", async () => {
-      const page = await newE2EPage({ html: snippet });
-      expect(
-        await page.evaluate(() => {
-          return window.getComputedStyle(document.body).getPropertyValue("--calcite-animation-timing");
-        })
-      ).toBe(" 300ms");
+    it("should set animation duration to 0ms when --animation-timing-factor set to zero", async () => {
```
Comment (jcfranco) [inline src/tests/globalStyles.e2e.ts:40]: For completeness, can you add a test that asserts on the default transition duration? You already cover disabling animation and not allowing component-level overrides.

**Review by jcfranco — APPROVED** [review]
Once the remaining items are covered, this is good to merge. Nice stuff, @anveshmekala! 🎉

**Comment by jcfranco** [comment]
@anveshmekala I think the PR title may need updating since this also introduces the animation timing factor variable.

**Pull request merged**